swarm/storage/resource.go

```diff
@@ -407,18 +407,38 @@ func (self *ResourceHandler) updateResourceIndex(rsrc *resource, chunk *Chunk, n
 	}
 
 	// update our rsrcs entry map
-	rsrc.lastBlock = nextblock
+	period, version, _, data, err := parseUpdate(chunk.SData[signatureLength:])
+	rsrc.lastPeriod = period
 	rsrc.version = version
-	rsrc.data = make([]byte, len(chunk.SData)-signatureLength)
 	rsrc.updated = time.Now()
-	copy(rsrc.data, chunk.SData[signatureLength:])
-	log.Debug("Resource synced", "name", rsrc.name, "key", chunk.Key, "block", nextblock, "version", version)
+	rsrc.data = make([]byte, len(data))
+	copy(rsrc.data, data)
+	log.Debug("Resource synced", "name", rsrc.name, "key", chunk.Key, "period", rsrc.lastPeriod, "version", rsrc.version)
 	self.resourceLock.Lock()
 	self.resources[*indexname] = rsrc
 	self.resourceLock.Unlock()
 	return rsrc, nil
 }
 
+func parseUpdate(blob []byte) (period uint32, version uint32, ensname []byte, data []byte, err error) {
```

Comment:
Aren't 6 return values a little too many? Maybe a struct type and an err would be better.

Reply:
it's 5 ;)

anyway, it's a private method, so I don't really see the harm.

Could use the `resource` object, but then the ensuing assignments will look strange because it will assign to fields in another resource object. Of course we could use a resource pointer and write to that directly, but that's a bit too `c` again ...

Maybe just keep it?